tools/c7n_gcp/c7n_gcp/resources/compute.py

```diff
@@ -478,6 +479,34 @@ def get(client, resource_info):
                         'snapshot': resource_info['snapshot_id']})
 
 
+@Snapshot.filter_registry.register('disk-availability')
+class CheckDiskAvailableSnapshotFilter(Filter):
+    """
+    That filter checks if the snapshot disk exists
+    """
+    schema = type_schema('disk-availability')
+    permissions = ('compute.disks.get',)
+
+    def process(self, resources, event=None):
+        session = local_session(self.manager.session_factory)
+        client = session.client(service_name='compute', version='v1', component='disks')
+        accepted_resources = []
+        # Getting project_id from client
+        project = session.get_default_project()
+        for resource in resources:
+            if 'sourceDisk' in resource:
```

Comment:
these should be grouped by source disk first, else we're doing redundant api, calls. its very common to have several snapshots on a single disk. we should only need to fetch a given disk once.

Reply:
I will check according to the existing policies

Reply:
thank you. Fixed